src/lib/weather-data.component.ts

```diff
@@ -1,73 +1,431 @@
-          tooltip: station.name,
-          icon: dwdStationIcon
+    /**
+     * when clicking a station, extracs vital information for html template.
+     * Resets usedDataType and usedResolution to not show outdated information
+     * @param station object clicked
+     */
+    prepareStationInformation(station: Station): void {
+        this.station = station;
+
+        this.activatedFilters.forEach((filter) => {
+            let a = station.capabilities
+                .filter((element) => element.dataType === filter)
+                .map((element) => element.resolution)
+                .sort();
+
+            this.availableResolutions.set(filter, a);
+            this.clearStationData();
+        });
+    }
+
+    /**
+     * clear all station information to not
+     * confuse them with new information
+     */
+    clearStationData(): void {
+        this.usedDataType = undefined;
+        this.usedResolution = undefined;
+        this.usedFrom = undefined;
+        this.usedUntil = undefined;
+    }
+
+    /**
+     * finds the lower and upper bound of time range
+     * of specified dataType and saves them to array
+     * @param dataType DataCapability to use
+     * @param resolution TimeResolution to use
+     */
+    createRangeOfTime(dataType: string, resolution: string): void {
+        if (this.station) {
+            this.station?.capabilities.find((element) => {
+                if (element.dataType === dataType && element.resolution === resolution) {
+                    this.availableTimeSlots[0] = this.formatTimestamp(element.availableFrom);
+                    this.availableTimeSlots[1] = this.formatTimestamp(element.availableUntil);
+                }
+            });
         }
-      }));
-    });
-  }
-
-  getWeatherDataByStation(): void {
-    this.weatherService.getWeatherDataByStation("/daily/qn/Oldenburg").subscribe(test => {
-      this.weatherData = test;
-      console.log(test);
-    })
-  }
-
-
-  /**
-  * get resolution data from dwd
-  */
-  getResolutions(): void {
-    this.weatherService.getResolutions('/').subscribe({
-      next: (response) => {
-        // extracts the meters content immediately, 
-        // so you dont have to do it all the time
-        this.possibleResolutions = response;
-        console.log(this.possibleResolutions);
-      },
-      error: (error) => {
-        console.log(error);
-      },
-    })
-  }
+    }
 
+    //------------------------------------------------------------------------------ Request weather data by station ----------------------------------
 
-}
+    setTypeAndResolution(dataType: string, resolution: string): void {
+        this.usedDataType = dataType;
+        this.usedResolution = resolution;
+    }
+
+    /**
+     * instructions for downloading the selected data.
+     * Uses global variables, thus it does not need parameters
+     * @returns if a global variable is not set -> the url can't be build
+     */
+    processWeatherInformation() {
```

Comment:
the function name does not match the functions actions. check if another name may be better suited

```suggestion
    startWeatherDataDownload() {
```